src/sidebar/components/annotation-action-bar.js

```diff
@@ -2,42 +2,70 @@ const propTypes = require('prop-types');
 const { createElement } = require('preact');
 
 const { withServices } = require('../util/service-context');
+const useStore = require('../store/use-store');
 const { isShareable, shareURI } = require('../util/annotation-sharing');
 
 const AnnotationShareControl = require('./annotation-share-control');
 const Button = require('./button');
 
 /**
- * A collection of `Button`s in the footer area of an annotation.
+ * A collection of `Button`s in the footer area of an annotation that take
+ * actions on the annotation.
  */
 function AnnotationActionBar({
   annotation,
-  onDelete,
+  annotationMapper,
+  flash,
   onEdit,
-  onFlag,
   onReply,
-  groups,
```

Comment:
Removal of both the `groups` and `session` service injection here in preference to getting at what is needed through the `store` instead. I presume this is the general pattern we'd like to follow—preference for using the store directly when possible—but wanted to call it out in case not.